types/engine/engine.go

```diff
@@ -77,12 +79,12 @@ type TransactionForCall struct {
 	Data     *primitives.VarData  `json:"data,omitempty"`
 }
 
-// UnmarshalJSON implements json.Unmarshaler
+// UnmarshalJSON implements jsoniter.Unmarshaler
```

Comment:
We couldn't fully remove `encoding/json` library because of some missing features of the `jsoniter`. Since both libraries coexist in the repo, I wanted to explicitly provide the library name to prevent careless maintenance changes.